packages/near-api-js/src/contract.ts

```diff
@@ -13,6 +16,52 @@ function nameFunction(name: string, body: (args?: any[]) => any) {
     }[name];
 }
 
+function validateArguments(args: object, abiFunction: AbiFunction, ajv: Ajv, abiRoot: AbiRoot) {
+    if (!isObject(args)) return;
+
+    if (abiFunction.params?.serialization_type !== AbiSerializationType.Json) {
+        throw new UnsupportedSerializationError(abiFunction.name, abiFunction.params.serialization_type);
```

Comment:
I missed that these get referenced in the error when I suggested optional chaining - these will need to be guarded as well to avoid throwing an exception while throwing an exception.

```suggestion
        throw new UnsupportedSerializationError(abiFunction.name, abiFunction.params?.serialization_type);
```

Reply:
Yeah, see my comment in the other thread. The flow was not actually supposed to reach this line if `params`/`result` is missing